src/util.ts

```diff
@@ -45,5 +45,8 @@ export function snakeToCamelCase(str: string) {
   if (splitted.length === 0) {
     return str;
   }
-  return [splitted[0], ...splitted.slice(1).map(capitalize)].join('');
+  return [
+    str.charAt(0) === '_' ? capitalize(splitted[0]) : splitted[0],
```

Comment:
As I mentioned in the doc, I believe the proper fix would be to make sure the leading `_` does not appear.  So in `camelToSnakeCase` above (line 22) the first capital letter should not have the `_` appended before it.

Reply:
To clarify: we still can leave this fix here as well, but let's also add similar logic to line 22.

Reply:
Would it make sense to simply have the fix in line 22 to remove the leading `_` and not have this fix here for simplicity?

Reply:
Synced offline, we keep the first position capitalization.
The fix in the line 22, add regex `/(?!^)[A-Z]/g`, as result, `IPProtocol` -> `I_p_protocol`.
sneak back to camel, will remain the same for the first position. as result, `I_p_protocol` -> `IPProtocol`